idmapping-rest/src/main/java/org/uniprot/api/idmapping/service/impl/PIRServiceImpl.java

```diff
@@ -53,6 +53,11 @@ public class PIRServiceImpl extends IdMappingPIRService {
                                     .getSearchFieldItemByName("accession_id")
                                     .getValidRegex()
                             + "(\\[[0-9]+\\-[0-9]+\\]|\\.[0-9]+)");
+    public static final Pattern UNIPROTKB_ACCESSION_REGEX =
+            Pattern.compile(
+                    SearchFieldConfigFactory.getSearchFieldConfig(UniProtDataType.UNIPROTKB)
```

Comment:
You can use FieldRegexConstants.UNIPROTKB_ACCESSION_REGEX.

Reply:
UNIPROTKB_ACCESSION_REGEX pattern is used in multiple places so it would be best to create a common function to get UNIPROTKB_ACCESSION_REGEX pattern. You can pass the regex to the function and get pattern in return.